File: backend/DoAn-CSharp/Models/Entities/POI.cs
using System;
using System.Collections.Generic;

namespace DoAn_CSharp.Models.Entities
{
    public class POI
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Slug { get; set; } = string.Empty;
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public int TriggerRadiusMeters { get; set; } = 30;
        public string Category { get; set; } = string.Empty; // restaurant, cafe, temple, market, park, landmark, street_art
        public int? CategoryId { get; set; }
        public POICategory? POICategory { get; set; }
        public int Priority { get; set; } = 5; // 1-10

        // Address
        public string? Address { get; set; }
        public string? Ward { get; set; }
        public string? District { get; set; }
        public string? City { get; set; }

        // Contact
        public string? Phone { get; set; }
        public string? Website { get; set; }
        public string? FacebookUrl { get; set; }

        public string? ImageUrl { get; set; }
        public string? GoogleMapsUrl { get; set; }
        public bool IsActive { get; set; } = true;
        public DateTime? DeletedAt { get; set; } // Soft Delete
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public int? OwnerId { get; set; }
        public Owner? Owner { get; set; }
        public string ApprovalStatus { get; set; } = "approved";
        public double Rating { get; set; } = 4.5;
        public int ReviewCount { get; set; } = 120; // pending, approved, rejected
        public string PriceRange { get; set; } = "1"; // "1" = Budget, "2" = Mid-range, "3" = Upscale
        public string? OperatingHours { get; set; }

        // Navigation properties
        public ICollection<POITranslation> Translations { get; set; } = new List<POITranslation>();
        public ICollection<MenuItem> MenuItems { get; set; } = new List<MenuItem>();
        public ICollection<QRCode> QRCodes { get; set; } = new List<QRCode>();
        public ICollection<VisitLog> VisitLogs { get; set; } = new List<VisitLog>();
        public ICollection<POIImage> Images { get; set; } = new List<POIImage>();
    }
}

File: frontend/src/types/poi.ts
export interface POI {
  id: number;
  name: string;
  slug: string;
  latitude: number;
  longitude: number;
  triggerRadiusMeters: number;
  category: string;
  categoryId?: number;
  priority: number;
  ownerId?: number;
  approvalStatus: string;
  address?: string;
  ward?: string;
  district?: string;
  city?: string;
  phone?: string;
  website?: string;
  facebookUrl?: string;
  imageUrl?: string;
  googleMapsUrl?: string;
  isActive: boolean;
  rating: number;
  reviewCount: number;
  localizedName: string;
  shortDescription: string;
  fullDescription: string;
  audioText: string;
  qrCode?: string;
  menuItemCount: number;
  distanceMeters?: number;
  isFavorite: boolean;
  priceRange: string;
  images: POIImage[];
  operatingHours?: string;
  createdAt: string;
  updatedAt: string;
}

export interface POIListItem {
  id: number;
  name: string;
  slug: string;
  latitude: number;
  longitude: number;
  category: string;
  imageUrl?: string;
  shortDescription: string;
  distanceMeters?: number;
  ownerId?: number;
  approvalStatus: string;
  isFavorite: boolean;
  rating: number;
  reviewCount: number;
  priceRange: string;
}

export interface POIImage {
  id: number;
  imageUrl: string;
  isCover: boolean;
}

export interface POICategory {
  id: number;
  slug: string;
  name: string;
  iconUrl?: string;
  color?: string;
  sortOrder: number;
  isActive: boolean;
}

export interface MenuItem {
  id: number;
  poiId: number;
  name: string;
  price: number;
  currency: string;
  imageUrl?: string;
  displayOrder: number;
  isAvailable: boolean;
  localizedName: string;
  localizedDescription: string;
}
